### PR TITLE
PV upgrade test

### PR DESCRIPTION
**Pull request by msau42**
**What this PR does / why we need it**:
This PR adds a PV upgrade test to the new upgrade test framework.  Before, this test had to be done manually.  Currently the upgrade test framework only works on the GCE environment, so I plan to add support for other providers later.  In order to write the test, I had to modify and refactor some volume test util libraries.  I reran the impacted tests to make sure they still passed.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
It's probably easier to review the two commits separately.  I split it up into the refactor changes, and the upgrade test changes.

**Release note**:

NONE

cc @saad-ali @krousey

### CODE REVIEW

File: test/e2e/upgrades/persistent_volumes.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}
+func (t *PersistentVolumeUpgradeTest) deleteGCEVolume(pvSource *v1.PersistentVolumeSource) {
+	framework.DeletePDWithRetry(pvSource.GCEPersistentDisk.PDName)
+}
+
+// Setup creates a pv and then verifies that a pod can consume it.  The pod writes data to the volume.
+func (t *PersistentVolumeUpgradeTest) Setup(f *framework.Framework) {
+
+	// TODO: generalize this to other providers
+	framework.SkipUnlessProviderIs("gce")
```
Comment (krousey) [inline test/e2e/upgrades/persistent_volumes.go]: Could this also work with GKE? If not, we have to still run this test manually for GKE.

File: test/e2e/upgrades/persistent_volumes.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+// PersistentVolumeUpgradeTest test that a pv is available before and after a cluster upgrade.
+type PersistentVolumeUpgradeTest struct {
+	pvSource *v1.PersistentVolumeSource
+	pv       *v1.PersistentVolume
+	pvc      *v1.PersistentVolumeClaim
+}
+
+const (
+	PVTestFile string = "/mnt/pv_upgrade_test"
```
Comment (krousey) [inline test/e2e/upgrades/persistent_volumes.go]: Any reason for these identifiers to be exported from the package?

File: test/e2e/upgrades/persistent_volumes.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	PVTestData string = "keep it pv"
+	PVWriteCmd string = "echo \"" + PVTestData + "\" > " + PVTestFile
+	PVReadCmd  string = "cat " + PVTestFile
+)
+
+func (t *PersistentVolumeUpgradeTest) createGCEVolume() *v1.PersistentVolumeSource {
```
Comment (krousey) [inline test/e2e/upgrades/persistent_volumes.go]: There's no need for this to be a method.

**Review by krousey — CHANGES_REQUESTED** [review]

File: test/e2e/upgrades/persistent_volumes.go (hunk 3 above)
Comment (msau42) [inline test/e2e/upgrades/persistent_volumes.go]: When I add AWS support, I think it will be better to separate them into separate methods.

**Comment by msau42** [comment]
@k8s-bot verify test this

**Comment by msau42** [comment]
@k8s-bot gce etcd3 e2e test this

**Review by krousey — APPROVED** [review]

**Pull request merged**